components/core/src/clp_s/search/Output.cpp

```diff
@@ -915,7 +915,9 @@ void Output::populate_string_queries(std::shared_ptr<Expression> const& expr) {
             }
 
             std::unordered_set<int64_t>& matching_vars = m_string_var_match_map[query_string];
-            if (query_string.find('*') == std::string::npos) {
+            if (query_string.find('*') == std::string::npos
+                && query_string.find('?') == std::string::npos)
```

Comment:
Similar to my other comment we should be handling escaping here as well. Specifically if there are no unescaped `*` or `?` we should enter this code block, process the string to unescape escaped characters as necessary, then do the precise search.

E.g. `\\var\?` should enter this code block and get unescaped to become `\var?` before performing exact match.

Reply:
I think the function `wildcard_match_unsafe_case_sensitive` can handle escape characters correctly. The corner case was mentioned above and will be fixed later.

Reply:
Right, but this code block uses `m_var_dict->get_entry_matching_value()` which doesn't respect escape characters.

Reply:
Maybe for the condition here we can add a method like
```
bool has_unescaped_wildchards(std::string const& query) {
    bool escaped{false};
    for (size_t i = 0; i < query.size(); ++i) {
        if (query[i] == '*' || query[i] == '?') {
            return true;
        }
        if (query[i] == '\') {
            ++i;
        }
    }
}
```

and replace the if condition with `false == has_unescaped_wildcards(query)`?